argparse_dataclass.py

```diff
@@ -369,8 +369,7 @@ def _add_dataclass_options(
             if len(f_args) == 2 and NoneType in f_args:
                 arg = next(a for a in f_args if a is not NoneType)
                 kwargs["type"] = arg
-            else:
-                raise TypeError("Union types other than 'Optional' are not supported")
+
```

Comment:
```suggestion
elif "type" not in kwargs:
                raise TypeError("Union types other than 'Optional' are not supported")
```
I think this should only be supported if type is supplied by the meta-data

Reply:
Maybe this whole section (lines 264-373) should be skipped if type is already defined by the meta-data.

Reply:
Thanks that you are willing to support Unions although you might never use it yourselves. I have adapted the PR to omit the block if "type" was specified. I agree it is better this way, since we get a more appropriate error this way.